modules/apps/commerce/commerce-currency-service/src/main/java/com/liferay/commerce/currency/internal/search/spi/model/query/contributor/CommerceCurrencyModelPreFilterContributor.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.currency.internal.search.spi.model.query.contributor;
+
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.filter.BooleanFilter;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.search.spi.model.query.contributor.ModelPreFilterContributor;
+import com.liferay.portal.search.spi.model.registrar.ModelSearchSettings;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Mahmoud Azzam
+ */
+@Component(
+	property = "indexer.class.name=com.liferay.commerce.currency.model.CommerceCurrency",
+	service = ModelPreFilterContributor.class
+)
+public class CommerceCurrencyModelPreFilterContributor
+	implements ModelPreFilterContributor {
+
+	@Override
+	public void contribute(
+		BooleanFilter booleanFilter, ModelSearchSettings modelSearchSettings,
+		SearchContext searchContext) {
+
+		_filterByType(booleanFilter, searchContext);
+	}
+
+	private void _filterByType(
+		BooleanFilter booleanFilter, SearchContext searchContext) {
+
+		String type = GetterUtil.getString(
+			searchContext.getAttribute(Field.TYPE));
```

Comment:
what is this field? I don't find it in the CommeceCurrencyModel defined in this PR. Can you please make sure this is right?